src/player-data.ts

```diff
@@ -147,37 +149,68 @@ export default class PlayerData extends EventEmitter {
    * @param power Power charged
    */
   chargePower(power: number) : number {
+    let brainstoneUsage = 0;
+
+    if (this.brainstone === BrainstoneArea.Area1 ) {
+      brainstoneUsage += 1;
+      power -= 1;
+      this.brainstone = BrainstoneArea.Area2;
+    };
+
     const area1ToUp = Math.min(power, this.power.area1);
-    const area2ToUp = Math.min(power - area1ToUp, this.power.area2 + area1ToUp );
+
+    if (this.brainstone === BrainstoneArea.Area2 && (power - area1ToUp)> 0) {
+      brainstoneUsage += 1;
+      power -=1;
+      this.brainstone = BrainstoneArea.Area3;
+    }
+
+    const area2ToUp = Math.min(power - area1ToUp , this.power.area2 + area1ToUp  );
 
     this.power.area1 -= area1ToUp;
     this.power.area2 += area1ToUp - area2ToUp;
     this.power.area3 += area2ToUp;
     
     //returns real charged power
-    return area1ToUp + area2ToUp;
+    return area1ToUp + area2ToUp + brainstoneUsage;
   }
 
   spendPower(power: number)  {  
+      const totPower = this.power.area3 + this.brainstoneValue() ; 
+      if (this.brainstone === BrainstoneArea.Area3 && ( power >= 3 || ( power<3 && this.power.area3<power))){
+        this.brainstone = BrainstoneArea.Area1;
+        power = Math.max(power-3,0);
+      }
       this.power.area3 -= power;
       this.power.area1 += power;
   }
 
   discardPower(power: number, type: Resource) {
     const area1ToGaia = Math.min(power, this.power.area1);
     const area2ToGaia = Math.min(power - area1ToGaia, this.power.area2);
-    const area3ToGaia = Math.min(power - area1ToGaia - area2ToGaia, this.power.area2);
+    const area3ToGaia = Math.min(power - area1ToGaia - area2ToGaia, this.power.area3);
+    const brainstoneNeeded = this.brainstoneInPlay() && (power - area1ToGaia - area2ToGaia - area3ToGaia >0);
```

Comment:
the second part can be `&& this.discardablePowerTokens() === power`.

Reply:
ok